### DISCUSSION TITLE
ESP-IDF Offline and Online install gets differents installation and both  does not work

### DISCUSSION THREAD

**Discussion by oldman152** [toitlang/jaguar#515] [Q&A]
I tried a lot of time to install the offline of ESP idf (esp-idf-tools-setup-offline-5.2.1.exe) and I get an incomplete installation (idf.py doesnot works because path is not configured for example)
When I try to intale the online version (esp-idf-tools-setup-online-2.24.exe) the install stop with the  code 101 because the link to the jaguar download does not existe anymore and even if I do it manually I do not know how to go on the install. 
below the end of the log:
##############################
"Done, exit code=101
2024-04-27 23:56:13.234   --------
2024-04-27 23:56:13.234   Running command: C:\Users\pc\AppData\Local\Temp\is-AO4MV.tmp\idf-env.exe toit reinstall --jaguar
                          Downloading https://github.com/toitlang/jaguar/releases/latest/download/jag_windows.zip to C:\EspT/dist/jag_windows.zi
                          Download o
                          Extracting to C:\Users\pc/AppData/Local/Programs/jagua
                          thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidArchive("Invalid zip header")', src\package.rs:20:5
                          stack backtrace"
#######################################
is some one can help me.
I want first to try to works with esp idf to run the debugger with esp prog.
Thank you for your help . JP

**Comment by floitsch** [comment]
The `jag_windows.zip` was renamed to `jag.zip`. I uploaded a copy of it under the old name. This should fix the fix the installation problem for the current release.
We will fix this more correctly next week.

**Comment by oldman152** [comment]
Thank you very much for your quick answer, 
I try to relunch the esp idf online install in the same directory but there is still a problem with jaguar, but not a download one.
here is the code : 
########################################################Running command: C:\Users\pc\AppData\Local\Temp\is-TK1PA.tmp\idf-env.exe toit reinstall --jaguar
                          Using cached archive: C:\EspT/dist/jag_windows.zi
                          Download o
                          Extracting to C:\Users\pc/AppData/Local/Programs/jagua
                          thread 'main' panicked at 'called `Result::unwrap()` on an `Err` value: InvalidArchive("Invalid zip header")', src\package.rs:20:5
                          stack backtrace
                             0:     0x7ff6f887f51c - git_tree_entry_filemode_ra
                             1:     0x7ff6f88a6ceb - git_tree_entry_filemode_ra
                             2:     0x7ff6f88794ef - git_tree_entry_filemode_ra
                             3:     0x7ff6f887f2cb - git_tree_entry_filemode_ra
                             4:     0x7ff6f8881d9a - git_tree_entry_filemode_ra
                             5:     0x7ff6f88819ef - git_tree_entry_filemode_ra
                             6:     0x7ff6f888244e - git_tree_entry_filemode_ra
                             7:     0x7ff6f888233d - git_tree_entry_filemode_ra
                             8:     0x7ff6f8880169 - git_tree_entry_filemode_ra
                             9:     0x7ff6f8882040 - git_tree_entry_filemode_ra
                            10:     0x7ff6f8a46685 - git_credential_username_ne
                            11:     0x7ff6f8a46b93 - git_credential_username_ne
                            12:     0x7ff6f86343be - <unknown
                            13:     0x7ff6f86020f2 - <unknown
                            14:     0x7ff6f8602aae - <unknown
                            15:     0x7ff6f86220bb - <unknown
                            16:     0x7ff6f8623ddd - <unknown
                            17:     0x7ff6f862455a - <unknown
                            18:     0x7ff6f8620567 - <unknown
                            19:     0x7ff6f86318a8 - <unknown
                            20:     0x7ff6f85fc38b - <unknown
                            21:     0x7ff6f86425b7 - <unknown
                            22:     0x7ff6f860d8ca - <unknown
                            23:     0x7ff6f85e94e6 - <unknown
                            24:     0x7ff6f8640c5c - <unknown
                            25:     0x7ff6f88723f8 - git_tree_entry_filemode_ra
                            26:     0x7ff6f860d9ec - <unknown
                            27:     0x7ff6f8a155fc - git_credential_username_ne
                            28:     0x7fff7aeb7344 - BaseThreadInitThun
                            29:     0x7fff7b2a26b1 - RtlUserThreadStar
                          
2024-04-28 18:09:41.044   --------
2024-04-28 18:09:41.044   Message box (OK):
                          Installation has failed with exit code 101
#########################################################

Thank you again for your help.jp
    Le dimanche 28 avril 2024 à 12:22:38 UTC+2, Florian Loitsch ***@***.***> a écrit :  
 
 


The jag_windows.zip was renamed to jag.zip. I uploaded a copy of it under the old name. This should fix the fix the installation problem for the current release.
We will fix this more correctly next week.

**Reply by floitsch** [reply]
Unfortunately I don't have access to a Windows machine right now.
I will try to reproduce tomorrow.

In the meantime: if you just want to install Jaguar, consider using `winget install --id=Toit.Jaguar -e` (for Windows 10+) or the Windows installer: https://github.com/toitlang/jaguar/releases/latest/download/jag_installer_x64.exe

**Comment by oldman152** [comment]
so thank you again, 

I will wait tomorrow, because I am not a good informatician, I have learnt C ad C++ a long time ago on motorola micropocessor, I now before becoming a retired I want to play with ESP32, and learn python, rust, ... to do some products for my pleasure. And I was stop by the ESP Prog debugger which doesnot work with the platformio alone ( I didnt find). Then I saw a youtuber who explained to install ESP-IDF to get the GDB and OPENOCD working well. And sinc 1 Months I stay at the same step. So thank you againif you help me to install ESP IDF.I do not know what is jaguar, and I just want to at the end of the installation to be sure before the nexte step.
But I have another question the install on and off line does not install the same thing, the on line seems to be more complete. Do you know why.
Best rergardsJPierre

    Le dimanche 28 avril 2024 à 20:50:42 UTC+2, Florian Loitsch ***@***.***> a écrit :  
 
 


Unfortunately I don't have access to a Windows machine right now.
I will try to reproduce tomorrow.

In the meantime: if you just want to install Jaguar, consider using winget install --id=Toit.Jaguar -e (for Windows 10+) or the Windows installer: https://github.com/toitlang/jaguar/releases/latest/download/jag_installer_x64.exe

**Reply by floitsch** [reply]
If you just want to install the ESP-IDF without Toit/Jaguar support, you can uncheck that box.
I'm guessing the offline version of the installer comes with certain, but not all, programs bundled, whereas the online version has support for more features it then dynamically downloads.
